app/src/main/java/com/google/android/sambadocumentsprovider/nativefacade/SambaProxyFileCallback.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.google.android.sambadocumentsprovider.nativefacade;
+
+import android.os.CancellationSignal;
+import android.os.ProxyFileDescriptorCallback;
+import android.system.ErrnoException;
+import android.system.StructStat;
+import android.util.Log;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+class SambaProxyFileCallback extends ProxyFileDescriptorCallback {
+    private static final String TAG = "SambaProxyFileCallback";
+
+    private final SambaProxyFileClient mFile;
+    private final ByteBuffer mBuffer;
+    private final CancellationSignal mSignal;
+
+    SambaProxyFileCallback(SambaProxyFileClient file, ByteBuffer buffer, CancellationSignal signal) {
+        mFile = file;
+        mBuffer = buffer;
+        mSignal = signal;
+    }
+
+    @Override
+    public long onGetSize() throws ErrnoException {
+        return mFile.mSize;
```

Comment:
We can discuss on this. File size might be changed along the way so I would see if it's possible to call stat() every time (unless this method is called too frequently that the performance is too poor).